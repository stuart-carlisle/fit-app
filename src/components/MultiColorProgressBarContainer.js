import React from 'react'
import MultiColorProgressBar from './MultiColorProgressBar'

export const MultiColorProgressBarContainer =({ data })=> {
    const target = data.target
    const usedEnergy = data.combinedTotal 
 
    const percentageEnergyUsed = Math.round((usedEnergy/target)*100)
    const percentageEnergyLeft = percentageEnergyUsed ? 100-percentageEnergyUsed : 100
    
    const readings = percentageEnergyLeft <= 0 ? 
    [
        {
            value: 100,
            color: '#ff0808'
        }
    ] :
    
    [
        {
            value: percentageEnergyUsed,
            color: '#4472C7'
        },
        {
            value: percentageEnergyLeft >= 100 ? 100 : percentageEnergyLeft,
            color: '#48e880'
        }
    ]
  
   
      return (
        <div>
            <MultiColorProgressBar readings={readings}/>
        </div>
      );
}
  
  
export default MultiColorProgressBarContainer;