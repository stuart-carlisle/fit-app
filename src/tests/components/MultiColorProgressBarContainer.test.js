import { shallow } from 'enzyme'
import React from 'react'
import MultiColorProgressBarContainer from '../../components/MultiColorProgressBarContainer'
 
test('Should render the multicolor progress bar container correctly',()=>{
    const data = {
        target: 2000,
        combinedTotal: 300
    }
    const wrapper = shallow(<MultiColorProgressBarContainer data={data} />)
    expect(wrapper).toMatchSnapshot()
})