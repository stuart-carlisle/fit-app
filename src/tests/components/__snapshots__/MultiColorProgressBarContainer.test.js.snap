// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the multicolor progress bar container correctly 1`] = `
<div>
  <MultiColorProgressBar
    readings={
      Array [
        Object {
          "color": "#4472C7",
          "value": 15,
        },
        Object {
          "color": "#48e880",
          "value": 85,
        },
      ]
    }
  />
</div>
`;
